api/v1alpha4/kubemarkmachine_types.go

```diff
@@ -53,7 +53,15 @@ type KubemarkMachineSpec struct {
 	KubemarkOptions KubemarkProcessOptions `json:"kubemarkOptions,omitempty"`
 
 	// KubemarkHollowPodClusterSecretRef is a reference to a secret with a kubeconfig for an external cluster used for kubemark pods.
+	// Deprecated: use BackingCluster instead; if both are defined, BackingCluster takes the precedence.
 	KubemarkHollowPodClusterSecretRef *corev1.ObjectReference `json:"kubemarkHollowPodClusterSecretRef,omitempty"`
+
+	// BackingCluster defined the cluster where to host the pod running kubemark.
+	// If empty, pods running kubemark will be created in the backing cluster defined in the KubemarkCluster resource;
+	// in case also KubemarkCluster.spec.backingCluster is empty or if the cluster uses a different infrastructure cluster kind like
+	// e.g. DockerCluster, pods running kubemark will be created in the management cluster, in the same namespace of the cluster resource.
+	// +optional
+	BackingCluster *BackingClusterSpec `json:"backingCluster,omitempty"`
```

Comment:
`BackingCluster` sounds a little ambiguous to me, maybe we could call it  `KubemarkPodHostCluster` ?

Reply:
@elmiko what about `ComputeCluster`?

Reply:
looking at the old value that this new approach will deprecate, we named it `KubemarkHollowPodClusterSecretRef`, what if we call this new field `KubemarkPodCluster` ?

i just don't want users to get confused when they see this, i feel like we have a couple references to different "clusters" and the topology can become confusing.

Reply:
I have the same concern, but when I looked at the old name my reaction was that this is exposing a lot of internals the user should not really care about (the fact that we run a binary that is called kubemark, the fact that this thing runs a Pod etc). 

It is also limiting because we are not creating only pods, but also secrets, and potentially more in the future, and (nit) having a field in a Kubemark* object that repeats kubemark kind of stutter 

This is is why I would prefer something that is easier to explain
- this is where the backend of my machines runs (BackingCluster)
- this is the thing that provides computing power to my machines (ComputeCluster)
- (???)

But I leave to you the final call, just tell me what do you pick

Reply:
i like your reasoning, i'm happy to go forward with ComputeCluster. thanks for explaining =)